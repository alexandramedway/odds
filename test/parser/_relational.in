do !true
do !false
do 1 == 1
do 41 != 42
do 41 < 42
do 42 <= 42
do 43 > 42
do 43 >= 42
do !(41 < 42)
do !(42 <= 42)
