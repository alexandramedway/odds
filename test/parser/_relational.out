[
    Do(Unop(Not, Bool_lit(true))) ;
    Do(Unop(Not, Bool_lit(false))) ;
    Do(Binop(Num_lit(1), Eq, Num_lit(1))) ;
    Do(Binop(Num_lit(41), Neq, Num_lit(42))) ;
    Do(Binop(Num_lit(41), Less, Num_lit(42))) ;
    Do(Binop(Num_lit(42), Leq, Num_lit(42))) ;
    Do(Binop(Num_lit(43), Greater, Num_lit(42))) ;
    Do(Binop(Num_lit(43), Geq, Num_lit(42))) ;
    Do(Unop(Not, Binop(Num_lit(41), Less, Num_lit(42)))) ;
    Do(Unop(Not, Binop(Num_lit(42), Leq, Num_lit(42))))
]
